### DISCUSSION TITLE
Roadmap for V3 (or next release)

### DISCUSSION THREAD

**Discussion by ldpr** [torrust/torrust-index-backend#93] [Q&A] [answered]
Hey guys and girls!

Is there a roadmap for when you guys might be cutting the next set of releases? I guess you might have something internal as it's not showing [here](https://github.com/orgs/torrust/projects/1).

If you don't that's totally fine, we're happy to wait - I'm just curious is all.
Thanks for everything ❤️

**Comment by da2ce7** [comment, marked as answer]
Hello @ldpr :)

We have not published a roadmap or plan for our next release.  Internally we have been working towards a consistent group-release for the Torrust Project: [Torrust Tracker](https://github.com/torrust/torrust-tracker), [Torrust Index](https://github.com/torrust/torrust-index) and the [Torrust Index GUI](https://github.com/torrust/torrust-index-gui).

This public release will include support for [upgrading form the previous version of the back-end](https://github.com/torrust/torrust-index/tree/develop/upgrades/from_v1_0_0_to_v2_0_0); Our goal with this group-release is to bake a consistent stable base for the community to upgrade to.

We expect that we will finish this release within two months.

Post-Release, we intend to have a community feedback and planing process to gather ideas and priorities from the users of our software, and together design a proper roadmap for the development of Torrust.

**Reply by ldpr** [reply]
Wonderful, thank you for the insight @da2ce7!

**Reply by da2ce7** [reply]
@ldpr long after two months... well 10months, we are still at work...

We have achieved much tho:

- [x] Good docker and container support.
- [x] Much more consistent codebase.
- [x] New [Torrust Website](https://torrust.com/).
- [x] Live [Testing Demo](https://tracker.torrust-demo.com/), please break and leave a nice bug reports 😜.

Overall, there is some more cleanup, going through the backlog of issues. But at least we are not far for a public beta release.

**Reply by josecelano** [reply]
- [x] Better documentation including tutorials on: https://torrust.com/blog
- [x] Intensive testing with thousands of real torrents
- [x] Benchmarking

Live demo URLs are:

- Torrust Index: https://index.torrust-demo.com/
- Torrust HTTP Tracker: https://tracker.torrust-demo.com/announce
- Torrust UDP Tracker: udp://tracker.torrust-demo.com:6969/announce

**Reply by ldpr** [reply]
> Intensive testing with thousands of real torrents

We'd be happy to help test, which do you think would be most useful? (~16k torrents)

* We can just test directly in production
* Setup a staging area and import data (less useful obviously because there won't be much usage apart from our testing)
* Can provide you a database to test with if you'd prefer to test yourself.

**Reply by josecelano** [reply]
Hi @ldpr what I did for testing was importing all torrents from https://academictorrents.com/ into the Index. I use this small importer script:

https://github.com/torrust/torrents-importer-sample

And there is a blog post here:

https://torrust.com/introducing-the-new-sample-torrent-migration-tool

You will need to re-write the code to import torrents from the source to your needs. If you don't have metadata (title, description, categories, tags) you can use any text. The title must be unique. You can use the info-hash, for example.

The script could be pretty annoying if too many torrents fail (I hope they don't :-)), because it's halted when an upload fails. You could change it to just store the list of failing torrents in a txt file or output.

You can use directly the [live demo](http://index.torrust-demo.com/) but unfortunately [yesterday I updated it with the latest develop branch tracker docker image and it's panicking](https://github.com/torrust/torrust-tracker/issues/582). I would wait until it's fixed although it should not affect torrents uploads. The only problem is the Index cannot whitelist the torrent (but we are using a public tracker) and the Index cannot get torrent statistics.

The demo is running in the smallest Digital Ocean droplet. It could be the case that we need to increase the size. I have not done load testing.

That would be very helpful but if you prefer it just send me the torrents and I'll do it.

In the future, I would like to have:

- A good set of test torrents with different cases.
- Improve the script to run the importation by using that set.
- Create a new app to test UDP trackers and HTTP trackers, like https://torrentdyne.com/

**Comment by josecelano** [comment]
Hi @ldpr, we have started working on the roadmap (WIP):

https://github.com/orgs/torrust/projects/10/views/1
